engine/src/main/java/org/hibernate/search/util/impl/ScopedAnalyzerReference.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.hibernate.search.analyzer.impl.AnalyzerReference;
+import org.hibernate.search.analyzer.impl.LuceneAnalyzerReference;
+
+/**
+ * A wrapper containing all the analyzers for a given class.
+ *
+ * @author Davide D'Alto
+ */
+public final class ScopedAnalyzerReference implements AnalyzerReference {
+
+	private final Map<String, AnalyzerReference> analyzers = new HashMap<>();
```

Comment:
That's never shared between threads, right? So we don't need to ensure safe publication and the like?